app/src/main/java/io/homeassistant/companion/android/onboarding/authentication/AuthenticationFragment.kt

```diff
@@ -155,21 +182,21 @@ class AuthenticationFragment : Fragment() {
             .setTitle(commonR.string.error_connection_failed)
             .setMessage(
                 when (sslError?.primaryError) {
-                    SslError.SSL_DATE_INVALID -> commonR.string.webview_error_SSL_DATE_INVALID
-                    SslError.SSL_EXPIRED -> commonR.string.webview_error_SSL_EXPIRED
-                    SslError.SSL_IDMISMATCH -> commonR.string.webview_error_SSL_IDMISMATCH
-                    SslError.SSL_INVALID -> commonR.string.webview_error_SSL_INVALID
-                    SslError.SSL_NOTYETVALID -> commonR.string.webview_error_SSL_NOTYETVALID
-                    SslError.SSL_UNTRUSTED -> commonR.string.webview_error_SSL_UNTRUSTED
+                    SslError.SSL_DATE_INVALID -> requireContext().getString(commonR.string.webview_error_SSL_DATE_INVALID)
+                    SslError.SSL_EXPIRED -> requireContext().getString(commonR.string.webview_error_SSL_EXPIRED)
+                    SslError.SSL_IDMISMATCH -> requireContext().getString(commonR.string.webview_error_SSL_IDMISMATCH)
+                    SslError.SSL_INVALID -> requireContext().getString(commonR.string.webview_error_SSL_INVALID)
+                    SslError.SSL_NOTYETVALID -> requireContext().getString(commonR.string.webview_error_SSL_NOTYETVALID)
+                    SslError.SSL_UNTRUSTED -> requireContext().getString(commonR.string.webview_error_SSL_UNTRUSTED)
                     else -> {
                         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                             when (error?.errorCode) {
                                 WebViewClient.ERROR_FAILED_SSL_HANDSHAKE ->
-                                    commonR.string.webview_error_FAILED_SSL_HANDSHAKE
-                                WebViewClient.ERROR_AUTHENTICATION -> commonR.string.webview_error_AUTHENTICATION
-                                WebViewClient.ERROR_PROXY_AUTHENTICATION -> commonR.string.webview_error_PROXY_AUTHENTICATION
-                                WebViewClient.ERROR_UNSUPPORTED_AUTH_SCHEME -> commonR.string.webview_error_AUTH_SCHEME
-                                WebViewClient.ERROR_HOST_LOOKUP -> commonR.string.webview_error_HOST_LOOKUP
+                                    requireContext().getString(commonR.string.webview_error_FAILED_SSL_HANDSHAKE)
+                                WebViewClient.ERROR_AUTHENTICATION -> requireContext().getString(commonR.string.webview_error_AUTHENTICATION)
+                                WebViewClient.ERROR_PROXY_AUTHENTICATION -> requireContext().getString(commonR.string.webview_error_PROXY_AUTHENTICATION)
+                                WebViewClient.ERROR_UNSUPPORTED_AUTH_SCHEME -> requireContext().getString(commonR.string.webview_error_AUTH_SCHEME)
+                                WebViewClient.ERROR_HOST_LOOKUP -> requireContext().getString(commonR.string.webview_error_HOST_LOOKUP)
                             }
                         }
                         message
```

Comment:
Found a bug, this will overwrite the specific error set by the error code above and should be placed in else branches. For example, on SSL handshake errors it will now show error code -11 instead of the specific handshake message.

Reply:
These strings are already in an `else` branch after all the SSL specific error messages. Is that what you mean? Just above the condition

Reply:
No it looks like inside this block it should also use an else? Without it when sending a `WebViewClient.ERROR_FAILED_SSL_HANDSHAKE` I'm getting the original message you provided, if I change it to something like this I correctly get the SSL handshake error:
```kotlin
else -> {
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
        when (error?.errorCode) {
           ... -> ...
           else -> message
        }
    } else {
        message
    }
}
```

Reply:
ooh gotcha, yea seems like it was always there. For some reason when I look at it the way it was before it looked _mostly_ correct lol

Reply:
usually android studio is good on calling out a missing else branch like that but seems it doesnt always complain when it should

Reply:
Maybe it was a Kotlin feature in that case we're not aware of? Doesn't matter in the end, it works as expected now.